lib/engine/tile.rb

```diff
@@ -63,15 +63,20 @@ def self.part(type, params, cache)
       case type
       when 'path'
         params = params.map do |k, v|
-          case v[0]
-          when '_'
-            [k, cache[v[1..-1].to_i]]
+          case k
+          when 'terminal'
+            [k, v.to_i]
           else
-            [k, Part::Edge.new(v)]
+            case v[0]
+            when '_'
+              [k, cache[v[1..-1].to_i]]
+            else
+              [k, Part::Edge.new(v)]
+            end
           end
         end.to_h
 
-        Part::Path.new(params['a'], params['b'])
+        Part::Path.new(params['a'], params['b'], params['terminal'])
```

Comment:
will need to add terminal: ...

Reply:
I'm afraid I don't understand, why won't params['terminal'] work?

Reply:
Are you saying we don't directly pass in the value for terminal, but rather the hash {terminal: value}?

Reply:
Nevermind. Keyword argument.